integration/query_frontend_test.go

```diff
@@ -161,6 +161,9 @@ func runQueryFrontendTest(t *testing.T, setup queryFrontendSetup) {
 
 	require.NoError(t, queryFrontend.WaitSumMetrics(e2e.Equals(numUsers*numQueriesPerUser), "cortex_query_frontend_queries_total"))
 
+	require.NoError(t, queryFrontend.WaitForPromQL(fmt.Sprintf("sum(cortex_request_duration_seconds_count) > %d", numUsers*numQueriesPerUser)))
+	require.NoError(t, querier.WaitForPromQL(fmt.Sprintf("sum(cortex_request_duration_seconds_count) > %d", numUsers*numQueriesPerUser)))
```

Comment:
Now that is way more powerful you can check for equality (`==` instead of `>`) filtering by the label `{route=''<query route>"}` where `<query route>` is the API v1 query route (don't remember the exact value in the label).